test/unit_tests/hoc_python/test_reference_counts.py

```diff
@@ -0,0 +1,281 @@
+import sys
+import neuron
+from neuron import h
+
+
+def test_section():
+    section = h.Section(name='section')
+    assert sys.getrefcount(section) == 2
+
+    # {Py_tp_repr, (void*) pysec_repr_safe},
+    repr(section)
+    assert sys.getrefcount(section) == 2
+
+    # {Py_tp_hash, (void*) pysec_hash_safe},
+    hash(section)
+    assert sys.getrefcount(section) == 2
+
+    #{Py_tp_call, (void*) NPySecObj_call_safe},
+    section()
+    assert sys.getrefcount(section) == 2
+
+    # {Py_tp_getattro, (void*) section_getattro_safe},
+    section.L
+    assert sys.getrefcount(section) == 2
+
+    section.Ra
+    assert sys.getrefcount(section) == 2
+
+    section.nseg
+    assert sys.getrefcount(section) == 2
+
+    section.rallbranch
+    assert sys.getrefcount(section) == 2
+
+    # not sure what the purpose of this is; return a dict of
+    # {'L': None, 'Ra': None, 'nseg': None, 'rallbranch': None}
+    d = section.__dict__
+    assert sys.getrefcount(d) == 2
+    assert sys.getrefcount(section) == 2
+
+    # {Py_tp_richcompare, (void*) pysec_richcmp_safe},
+    section == section
+    section != section
+    section < section
+    section <= section
+    section > section
+    section >= section
+    assert sys.getrefcount(section) == 2
+
+    # {Py_tp_iter, (void*) seg_of_section_iter_safe},
+
+    # {Py_tp_methods, (void*) NPySecObj_methods},
+    dir(section)
+    assert sys.getrefcount(section) == 2
+
+    # {Py_tp_doc, (void*) "Section objects"},
+    help(section)
+    assert sys.getrefcount(section) == 2
+
+    # {Py_sq_contains, (void*) NPySecObj_contains_safe},
+    "foo" in section
+    assert sys.getrefcount(section) == 2
+
+    #static PyType_Slot nrnpy_AllSegOfSecIterType_slots[] = {
+    it = section.allseg()
+    assert sys.getrefcount(it) == 2
+    assert sys.getrefcount(section) == 3
```

Comment:
```suggestion
    assert sys.getrefcount(section) == base_refcount + 1
```
The `+1`, because we're holding one additional reference.